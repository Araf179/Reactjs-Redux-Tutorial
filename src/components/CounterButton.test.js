import {shallow, mount, render} from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

it("Expect to render counterbutton", () => {
    const mockColor = 'red';
    expect(shallow(<CounterButton color={mockColor}/>).length).toMatchSnapshot()
})

it("correctly increments the counter", () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor}/>)

    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count : 1});
    wrapper.find('[id="counter"]').simulate('keypress');
    expect(wrapper.props().color).toEqual('red')
})