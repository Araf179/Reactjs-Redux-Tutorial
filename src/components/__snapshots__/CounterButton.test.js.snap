// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render counterbutton 1`] = `1`;
